react/features/polls/functions.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+/**
+ * Filter out duplicate choices.
+ *
+ * @param {Object} choices - Object containing choices listed by ID.
+ * @returns {Object}
+ */
+export function getUniquePollChoices(choices: Object): Object {
+    const newObj = {};
+    const seen = new Set<string>();
+
+    for (const key in choices) {
+        if (Object.prototype.hasOwnProperty.call(choices, key)) {
+            const text = choices[key].text.trim();
+
+            if (!seen.has(text) && text !== '') {
+                newObj[key] = choices[key];
+                seen.add(text);
+            }
+        }
+    }
+
+    return newObj;
+}
```

Comment:
Arguably if this isn't going to be re-used it can be stuffed in the file that uses it and moved out later when it needs to be reused. Unless you're already thinking about the mobile case.

This feature will have to be developed on mobile as well. Are you game? If not let us know.

Reply:
Let me restart the last part for clarity: the mobile side is a requirement. You can do it if you want as a separate endeavor, but this PR will not be blocked on it. This PR also will not be blocked on finalized design, although that might mean not enabling it by default on jitsi deployments, but I will try to sync with some people about UI/UX changes that should be included.

Reply:
I have no problem to continue working on the mobile feature. Yes I prefer to make it as a separate PR. Just want to clarify that this may be delayed due to my exams.